### PR TITLE
fix: Ensure config keys are printed for config errors

### LINKED ISSUE #17960: Change Request: Provide config key when config validation fails
### ESLint version

8.55.0

### What problem do you want to solve?

When the config file fails validation there is very little information to go on. In connection with #17959, using `"error"` for `reportUnusedDisableDirectives` causes the load to fail with error below which has no information on which config caused the error or call stack entry with line numbers.

```
% npm run lint -- src/www/main.js

> vite-poc@0.0.0 lint
> eslint src/www/main.js


Oops! Something went wrong! :(

ESLint: 8.55.0

TypeError: Expected a Boolean.
    at Object.boolean [as validate] (/Users/marty/gitmllc/vite-poc/node_modules/@humanwhocodes/object-schema/src/validation-strategy.js:36:19)
    at ObjectSchema.validate (/Users/marty/gitmllc/vite-poc/node_modules/@humanwhocodes/object-schema/src/object-schema.js:285:35)
    at Object.validate (/Users/marty/gitmllc/vite-poc/node_modules/@humanwhocodes/object-schema/src/object-schema.js:180:32)
    at ObjectSchema.validate (/Users/marty/gitmllc/vite-poc/node_modules/@humanwhocodes/object-schema/src/object-schema.js:285:35)
    at /Users/marty/gitmllc/vite-poc/node_modules/@humanwhocodes/object-schema/src/object-schema.js:239:18
    at Array.reduce (<anonymous>)
    at ObjectSchema.merge (/Users/marty/gitmllc/vite-poc/node_modules/@humanwhocodes/object-schema/src/object-schema.js:237:24)
    at /Users/marty/gitmllc/vite-poc/node_modules/@humanwhocodes/config-array/api.js:966:42
    at Array.reduce (<anonymous>)
    at FlatConfigArray.getConfig (/Users/marty/gitmllc/vite-poc/node_modules/@humanwhocodes/config-array/api.js:965:39)
```

### What do you think is the correct solution?

Report:
- the configuration setting that failed validation
- the found value for the setting that failed validation
- line number would be nice

### Participation

- [X] I am willing to submit a pull request for this change.

### Additional comments

Is there a type definition for new config file? I could see implementing this as type validation as well.

### PR DESCRIPTION
**Pull request by nzakas**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[x] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

Upgraded `config-array` to pull in the new `object-schema` and added a test for the CLI output.

refs #17964
fixes #17960

### CODE REVIEW

**Review by fasttime — COMMENTED** [review]
LGTM, thanks!

**Review by snitin315 — APPROVED** [review]
LGTM, thanks!

Do we want to add tests for https://github.com/eslint/eslint/issues/17964 in the same PR?

**Review by fasttime — APPROVED** [review]

File: tests/lib/config/flat-config-array.js [hunk 1]
```diff
@@ -2264,4 +2264,27 @@ describe("FlatConfigArray", () => {
         });
 
     });
+
+    describe("invalid options", () => {
+
+        it("should report key when an invalid option is found during merge", () => {
```
Comment (bmish) [inline tests/lib/config/flat-config-array.js:2270]: Isn't this the same as the above `should error when an unexpected value is found` test that changed?

**Review by bmish — APPROVED** [review]

**Comment by nzakas** [comment]
@snitin315 I was going to leave that for @mdjermanovic to do separately so we can get this into the next release.

File: tests/lib/config/flat-config-array.js (hunk 1 above)
Comment (nzakas) [reply tests/lib/config/flat-config-array.js:2270]: Oops, yes, thought I had removed this. Thanks.

**Pull request merged**